prusti-interface/src/specs/mod.rs

```diff
@@ -343,12 +361,23 @@ impl<'a, 'tcx> intravisit::Visitor<'tcx> for SpecCollector<'a, 'tcx> {
             let spec_item = SpecItem {spec_id, spec_type, specification};
             self.spec_items.push(spec_item);
 
-            // Collect loop invariant
+            // Collect invariants
             if spec_type == SpecType::Invariant {
-                self.loop_specs
+                if is_loop_invariant {
+                  self.loop_specs
```

Comment:
The indentation is wrong, but also this branching should be replaced by the two separate `SpecType`s mentioned above.